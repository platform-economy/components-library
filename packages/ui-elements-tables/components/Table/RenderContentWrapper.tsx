import React, { useEffect } from 'react';

export type RenderContentWrapperProps = {
  children: React.ReactNode;
  getHeight: (height: number) => void;
};

export const RenderContentWrapper = (props: RenderContentWrapperProps) => {
  const contentRef = React.createRef<HTMLDivElement>();

  const getHeight = () => {
    const node = contentRef.current;
    /* istanbul ignore else */
    if (node) {
      props.getHeight(node.getBoundingClientRect().height);
    }
  };

  useEffect(() => {
    getHeight();
  });

  return (
    <div ref={contentRef}>
      {props.children}
    </div>
  );
};
