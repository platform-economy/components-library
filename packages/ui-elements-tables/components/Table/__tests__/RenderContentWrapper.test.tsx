import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';

import 'jest-styled-components';

configure({ adapter: new Adapter() });

// Component
import { RenderContentWrapper } from '../RenderContentWrapper';

const getHeight = jest.fn(() => null);

describe('ui-elements-tables', () => {
  describe('Components', () => {
    describe('RenderContentWrapper', () => {
      it('Should render without throwing an exception', () => {
        const tree = TestRenderer
          .create(
            <RenderContentWrapper getHeight={getHeight}>
              content
            </RenderContentWrapper>,
          )
          .toJSON();

        expect(tree).toMatchSnapshot();
      });

      it('Should call getHeight function after mount', () => {
        mount(
          <RenderContentWrapper getHeight={getHeight}>
            content
          </RenderContentWrapper>,
        );

        expect(getHeight).toBeCalledTimes(1);
      });
    });
  });
});
