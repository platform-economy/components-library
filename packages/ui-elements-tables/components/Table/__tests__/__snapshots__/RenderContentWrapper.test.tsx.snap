// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-tables Components RenderContentWrapper Should render without throwing an exception 1`] = `
<div>
  content
</div>
`;
